Engine/include/VGDAssert.hpp

```diff
@@ -0,0 +1,86 @@
+//
+// VGDAssert.hpp
+// VGD Engine
+//
+// Created by Benjamin Arnold on 29 Aug 2015
+//
+
+/*! \file VGDAssert.hpp
+* @brief Custom throw assertion for VGD.
+*/
+
+#pragma once
+
+
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#include <exception>
+
+// https://www.softwariness.com/articles/assertions-in-cpp/
```

Comment:
Source code from that website is BSD 3-Clause:

Example source code is provided on the Softwariness Github site under a 3-clause BSD-style license, which you will see in source file headers or LICENSE files in the Github repositories. This means that you can use the example code for personal or commercial use provided you stick to the license. Sometimes, third-party code may be included that is under a different license, and you should check for that in the LICENSE files if you intend to redistribute.